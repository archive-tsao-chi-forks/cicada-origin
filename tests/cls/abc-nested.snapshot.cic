def the_nat_type(_: Trivial): Type { Nat }

class ABC { a: Trivial, b: the_nat_type(a), c: String }

def abc: ABC = {
  a: sole, b: 2, c: "c"
}

def xyz: class {
  x: String,
  y: String,
  z: ABC
} = {
  x: "x",
  y: "y",
  z: abc,
}

xyz

xyz.z
xyz.z.a
xyz.z.b
xyz.z.c
