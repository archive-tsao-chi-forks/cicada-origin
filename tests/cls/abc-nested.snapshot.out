the class  {
  x: String
  y: String
  z: class ABC {
    a: Trivial
    b: Nat
    c: String
  }
} {
  x: "x"
  y: "y"
  z: {
    a: sole
    b: 2
    c: "c"
  }
}
the class ABC {
  a: Trivial
  b: Nat
  c: String
} {
  a: sole
  b: 2
  c: "c"
}
the Trivial sole
the Nat 2
the String "c"

